### PR TITLE
CBL-5947: Merge Beryllium into master

### PR DESCRIPTION
**Pull request by callumbirks**
The merge is not as bad as I thought it would be. A few conflicts needed resolving, but no manual adjustment needed to take place. I've run the VectorSearch tests locally and they are passing.

### CODE REVIEW

**Comment by callumbirks** [comment]
Blackduck failure is a strange one - seems to have trouble with some mysterious character in the git log output. I ran the same git log command locally and used grep, and then manual looking, and could not find this mysterious character, so I think we can ignore it.

**Comment by callumbirks** [comment]
Blocked by https://github.com/couchbase/couchbase-lite-core-EE/pull/43

**Comment by callumbirks** [comment]
Reverted "Give CMake build -Werror" because it causes many errors in linux build. As its not a simple 4-line change, going to create a separate ticket for that.

**Comment by callumbirks** [comment]
@jianminzhao If you could give this a review now, the only failure is a strange error from the blackduck manifest check which seems to have a problem with some non-ASCII character in the `git log` output (which I couldn't find)

File: Crypto/PublicKey+Apple.mm [hunk 1]
```diff
@@ -539,12 +540,12 @@ virtual int _sign(int/*mbedtls_md_type_t*/ mbedDigestAlgorithm,
             
             SecTrustResultType result; // Result will be ignored.
             OSStatus err;
-            CFErrorRef cferr;
-            if (!SecTrustEvaluateWithError(trustRef, &cferr)) {
-                auto error = (__bridge NSError*)cferr;
-                LogVerbose(TLSLogDomain, "SecTrustEvaluateWithError failed: %s", error.description.UTF8String);
-            }
-            err = SecTrustGetTrustResult(trustRef, &result);
+                CFErrorRef cferr;
```
[the remark this answers is not in the document]
Comment (jianminzhao) [reply Crypto/PublicKey+Apple.mm:543]: and it passed formatting.

**Comment by borrrden** [comment]
> @jianminzhao If you could give this a review now, the only failure is a strange error from the blackduck manifest check which seems to have a problem with some non-ASCII character in the `git log` output (which I couldn't find)

Change ascii to utf-8 and it should probably be fine

File: Crypto/PublicKey+Apple.mm (hunk 1 above)
[the remark this answers is not in the document]
Comment (callumbirks) [reply Crypto/PublicKey+Apple.mm:543]: Not sure why. clang-format doesn't work for Obj-C (I don't think). So formatting is not run on this file.

**Comment by borrrden** [comment]
Now the big question:  Which parts of this were conflicts that need to be reviewed and which parts applied cleanly?

**Review by jianminzhao — APPROVED** [review]

**Comment by callumbirks** [comment]
> Now the big question:  Which parts of this were conflicts that need to be reviewed and which parts applied cleanly?

@borrrden Well, there were roughly 12 files with conflicts IIRC, each with about 3 conflicts on average. Almost all of those conflicts were ones where I just needed to pick between the two, because they're different versions of the same thing. So this meant mostly choosing the version from 3.2.

If we think it needs it, I could go through the conflicts again in a meeting?

**Comment by borrrden** [comment]
Excellent, so no conflicts that were too complicated? That sounds good enough to me

**Review by borrrden — APPROVED** [review]

**Pull request merged**